kernel/src/main/java/org/kframework/parser/binary/BinaryParser.java

```diff
@@ -53,72 +57,100 @@
  */
 public class BinaryParser {
 
-    private static final byte[] MAGIC = {0x7f, 'K', 'A', 'S', 'T'};
+    public static final byte[] MAGIC = {0x7f, 'K', 'A', 'S', 'T'};
 
     public static final int BEGIN = 0, KTOKEN = 1, KAPPLY = 2, KSEQUENCE = 3, KVARIABLE = 4, KREWRITE = 5,
-            INJECTEDKLABEL = 6, END = 7;
+            INJECTEDKLABEL = 6, END = 7, BACK_REFERENCE = 8;
```

Comment:
here I am decreasing the size of binary configurations on disk by adding some sharing.